### PR TITLE
Allow Deploying SageMaker Endpoints to Another AWS Account #4921

### PR DESCRIPTION
**Pull request by andresionek91**
## What changes are proposed in this pull request?

Currently, MLFlow will only allows deploying SageMaker endpoints to the currently active AWS account. This feature request aims to allow deployment to a different account by assuming a cross-account role.

## How is this patch tested?

Created unit tests to verify if:
* Assume role operation is executed successfully
* Deployment with assume role works
* Deploy with non existent assume role fails

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Allow assuming a role in another AWS Account to deploy SageMaker Endpoints.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [x] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: dev/small-requirements.txt [hunk 1]
```diff
@@ -12,3 +12,4 @@ pytest-localserver==0.5.0
 moto!=2.0.7
 azure-storage-blob>=12.0.0
 azure-identity>=1.6.1
+sure==2.0.0
```
Comment (dbczumar) [inline dev/small-requirements.txt:15]: @andresionek91 Can we drop the dependency on `sure` for testing? I think it may be causing some CI-related issues.

**Review by dbczumar — APPROVED** [review]
@andresionek91 Apologies for the delay. This looks great! Looks like there are some small linter errors from https://github.com/mlflow/mlflow/runs/4215347474?check_suite_focus=true, and I left one comment about dropping the `sure` dependency: https://github.com/mlflow/mlflow/pull/4923/files#r749671303. Very excited to merge this once these comments are addressed!

**Comment by andresionek91** [comment]
@dbczumar the linting issues + sure dependency are fixed. Can you please approve the workflows?

**Comment by andresionek91** [comment]
@dbczumar all tests passing. Thanks!

**Comment by dbczumar** [comment]
@andresionek91 Merged! Thank you for your contribution!

**Pull request merged**